### DISCUSSION TITLE
Message header: mehrere service requests in einer KIM-Nachricht?

### DISCUSSION THREAD

**Discussion by bNaumann-gem** [gematik/spec-E-Rezept-ServiceRequest#63] [Q&A] [answered]
**Q:** Können in einer KIM-Nachrichtig mehrere service request (eRP-Anforderungen) versendet werden oder wird erwartet, dass je KIM nur ein SR erfolgt?

**Je KIM nur ein SR (1:1)**
+ rezeptausstellendes System kann die Anfrage einfach verarbeiten
- Je eRP ist eine KIM erforderlich

**Je KIM können n SR geliefert werden (1:n)**
- rezeptausstellendes System muss alle SRs technisch (über ein Script) verarbeiten können
+ es können beliebig viele eRPs angefordert werden

(!) die technische Architektur musss berücksichtigen, alles in einem Bundle darzustellen, aktuell wird dies noch nicht vorgegeben.

- Jeder SR enthält bereits eine eindeutige request-ID
- Alle SRs können aber über eine requisition-ID "verbunden" werden
Siehe https://simplifier.net/erezept-servicerequest/erp-service-request-prescription-request

Klärung in eRP-in-KH-Sprechstunde am 30.11.2023

**Comment by SvenSommer** [comment]
Wie in der eRP-in-KH-Sprechstunde am 30.11.2023 besprochen, möchten wir mit einem Hinweis auf diesen (gewollten) Umstand aufmerksam machen. 
Um es zukünftigen Implementierern leichter zu machen:
https://github.com/gematik/spec-E-Rezept-ServiceRequest/pull/52

**Comment by florianschoffke** [comment]
Implementation Guide aktualisiert

**Reply by bNaumann-gem** [reply]
Kannst du die Antwort hier nochmal einfügen?

**Comment by omeiners** [comment]
Der Implementation Guide 'Kommunikation für das E-Rezept' schreibt unter "Einführung" - "Allgemeine Festlegungen" - "Anzahl von E-Rezepten pro Nachricht" vor, dass eine Nachricht nur ein Rezept adressiert. Dieser Abschnitt muss noch angepasst werden.

**Comment by florianschoffke** [comment, marked as answer]
Eine Beschreibung im IG kommt mit https://github.com/gematik/spec-E-Rezept-ServiceRequest/pull/67.
Zu Klärung der ursprünglichen Frage hier eine Antwort:

## Aufbau der Nachricht
Generell ist es so zu verstehen, dass eine Nachricht ein Message Bundle ist.
Dieses Message Bundle enthält als ersten Eintrag den Message Header. Dieser beauskunftet Sender und Empfänger der Nachricht und enthält den entsprechenden Event-Code, der eindeutig identifiziert, um welche Nachricht es sich im Anwendungsfall handelt.

## Identifikation von Rezeptanforderungen
Im MessageHeader gibt es das Feld `focus`. Hier wird auf die jeweiligen Rezeptanforderungen (ServiceRequest-Objekte) verwiesen. Eine Nachricht, die bspw. zwei Rezeptanforderungen enthält hat:
* Genau einen MessageHeader
* Genau zwei Referenzen in MessageHeader.focus
* Auf genau zwei ServiceRequest-Objekte im Bundle
* Mehrere Objekte im Bundle, die als Information für die ServiceRequests dienen

## Vorgeschlagene Logik in der Verarbeitung
Ein empfangendes System sollte den MessageHeader auslesen und ermitteln können wie viele Rezeptanforderungen (ServiceRequest-Objekte) in `MessageHeader.focus` referenziert sind.
Jeder ServiceRequest enthält alle Informationen zu genau einer Rezeptanforderung. D.h. es ist eine "Trägerressource", um alle Informationen zur Darstellung zu bringen.

## Zusammenfassung
Also jede Nachricht hat 1:n Rezeptanforderungen und jede Rezeptanforderung fragt 1:1 Rezepte an.
📝: Für mehrere ServiceRequests in einer Nachricht kann der gleiche Identifier unter `ServiceRequest.requisition` angegeben werden. Danach lassen sich die ServiceRequests einer "Anfrage" zuordnen.

Folgende Darstellung soll die Beschreibung verdeutlichen:
<img width="827" alt="Screenshot 2024-03-18 at 14 03 05" src="https://github.com/gematik/spec-E-Rezept-ServiceRequest/assets/24414959/8690b8d1-a36b-465e-ae7a-2d1900da1c37">
